src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -50,6 +52,54 @@ export default class SimidMapCreative extends BaseSimidCreative {
    * @private 
   */
   grantLocationAccess_() {
-    //ToDo(kristenmason@): implement map
+    //ToDo(kristenmason@): implement the Google Maps request access functionality
+    try{
+      this.simidProtocol.sendMessage(CreativeMessage.REQUEST_PAUSE).then(() => {
+        findNearest.classList.add("hidden");})
+    }
+    catch(exception){
```

Comment:
Instead of using try catch for a promise, there are two ways.

Either make the second parameter a function for when the promise rejects (less good).

Or chain them.
this.simidProtocol.sendMessage(CreativeMessage.REQUEST_PAUSE).then(
    () => {
        findNearest.classList.add("hidden");
    }).catch(
    () => {
     /// whatever you're going to do you can either fatal error, or you can log a message.
   });


Here's a good article you could read.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises

Reply:
thanks for the link!